src/theory/sets/cardinality_extension.cpp

```diff
@@ -1028,7 +1028,7 @@ void CardinalityExtension::mkModelValueElementsFor(
           // the current members of this finite type.
 
           Node slack = nm->mkSkolem("slack", elementType);
-          Node singleton = nm->mkNode(SINGLETON, slack);
+          Node singleton = nm->mkSingleton(elementType, slack);
```

Comment:
This is safe because `elementType` is extracted from the set eqc which contains the constructed slack element.